pkg/cluster/internal/providers/podman/provision.go

```diff
@@ -307,6 +303,11 @@ func generatePortMappings(clusterIPFamily config.ClusterIPFamily, portMappings .
 		// generate the actual mapping arg
 		protocol := string(pm.Protocol)
 		hostPortBinding := net.JoinHostPort(pm.ListenAddress, fmt.Sprintf("%d", pm.HostPort))
+		// Podman expects empty string instead of 0 to assign a random port
```

Comment:
:man_shrugging:

Reply:
Yeah ... 🤷‍♂️
This is why we needed a provider and not just a switch between calling docker and podman. They're not actually 1:1 compatible and podman has a number of bugs we need to avoid (mostly currently by requiring a new enough version ...)